widget/view.controller.js

```diff
@@ -2,486 +2,613 @@
-        if (result && result['hydra:member'] && result['hydra:member'].length > 0) {
-          let dataToPlot = result['hydra:member'];
-          createDataToPlot(dataToPlot);
-        } else {
-          errorMessage = $scope.viewWidgetVars.MESSAGE_NO_RECORDS_FOUND;
-          renderNoRecordMessage();
-        }
-      }, function () {
-        errorMessage = $scope.viewWidgetVars.MESSAGE_NO_RECORDS_FOUND;
-        renderNoRecordMessage();
-      }).finally(function () {
-        $scope.processing = false;
-      });
-    }
-
-    //create nodes and links data required to plot on Sankey chart
-    function createDataToPlot(dataToPlot) {
-      let idIndex = 0;
-      let nodesMap = [];
-      let seriesArray = [];
-      noOfSeries = $scope.config.layers.length + 1;
-      let seriesNameArray = [];
-      backgroundArea = [];
-      //replace null values in response data with 'Unknown' as value
-      var replacedArray = _.map(dataToPlot, function (obj) {
-        return _.mapObject(obj, function (value, key) {
-          return value === null ? "Unknown" : value;
-        });
-      });
-      dataToPlot = replacedArray;
-      //create series array as per the aggregate of series received in reponse data
-      for (var i = 0; i < noOfSeries; i++) {
-        seriesNameArray.push(['series_' + i]);
-        seriesArray.push(_.filter(_.map(dataToPlot, item => item['series_' + i]), (item, index, arr) => item !== null && arr.indexOf(item) == index));
-      };
-      seriesArray.forEach((element, s) => {
-        if (element.length > 0) {
-          if ($scope.config.moduleType === 'Across Modules' && $scope.config.layers[s]) {
-            backgroundArea.push({ 'label': $scope.config.layers[s]['label'] });
+  'use strict';
+  (function () {
+    angular
+      .module('cybersponse')
+      .controller('socOverviewSankey210Ctrl', socOverviewSankey210Ctrl);
+  
+    socOverviewSankey210Ctrl.$inject = ['$scope', '$rootScope', 'config', '_', 'PagedCollection', 'widgetUtilityService', 'socOverviewSankeyService'];
+  
+    function socOverviewSankey210Ctrl($scope, $rootScope, config, _, PagedCollection, widgetUtilityService, socOverviewSankeyService) {
+  
+      var sankey;
+      var chartData = {};
+      var nodes = [];
+      var links = [];
+      var nodesMap = [];
+      var nodeColorMap = [];
+      var backgroundArea = [];
+      var errorMessage = '';
+      var _seriesNo = 0;
+      var _linkNo = 0;
+      var _linkId = 0;
+      var dataToPlot = [];
+      var idIndex = 0;
+      $scope.config = config;
+      $scope.moduleType = $scope.config.moduleType;
+      $scope.processing = false;
+      $scope.duration = 90;	// Default duration
+      $scope.config.buttons = [{
+        id: 'btn-6m',
+        text: 'BUTTON_LAST_6_MONTHS',
+        onClick: function () {
+          refreshSankey(90, 'btn-6m');
+        },
+        active: true,
+        type: 'submit'
+      }, {
+        id: 'btn-3m',
+        text: 'BUTTON_LAST_3_MONTHS',
```

Comment:
use `$scope.viewWidgetVars.BUTTON_LAST_3_MONTHS` to access button text for all buttons